File: CS/PS/Site/programmers/lv2/튜플.js
function solution(s) {
  var answer = [];
  let parse = "";
  let len = s.length;
  for (let i = 0; i < len; i++) {
    if (s[i] === "{") {
      parse += "[";
    } else if (s[i] === "}") {
      parse += "]";
    } else parse += s[i];
  }

  let parseArr = JSON.parse(parse);
  let sorted = parseArr.sort((a, b) => {
    return a.length - b.length;
  });
  let set = new Set();
  for (let i = 0; i < sorted.length; i++) {
    for (let j = 0; j < sorted[i].length; j++) {
      set.add(sorted[i][j]);
    }
  }

  return [...set];
}

console.log(solution("{{2},{2,1},{2,1,3},{2,1,3,4}}"));
console.log(solution("{{1,2,3},{2,1},{1,2,4,3},{2}}"));
console.log(solution("{{123}}"));
console.log(solution("{{20,111},{111}}"));
console.log(solution("{{4,2,3},{3},{2,3,4,1},{2,3}}"));

File: CS/PS/Site/programmers/기출문제/19kakaoWinterIntern/튜플.py
import json
def solution(s):
    answer = []
    candi = []
    lst = []
    idx = 0
    json_str = ""
    
    while idx != len(s):
        if s[idx] == '{':
            json_str += '['
        elif s[idx] == '}':
            json_str += ']'
        else :
            json_str += s[idx]
        idx += 1
    lst = json.loads(json_str)
    lst.sort(key=len)
    dic = {}
    for r in lst:
        for v in r:
            if v not in dic:
                answer.append(int(v))
                dic[v] = 1
    return answer